workflows/sarek.nf

```diff
@@ -56,7 +63,129 @@ def checkPathParamList = [
 for (param in checkPathParamList) if (param) file(param, checkIfExists: true)
 
 // Set input, can either be from --input or from automatic retrieval in WorkflowSarek.groovy
-input_sample = params.build_only_index ? Channel.empty() : extract_csv(file(WorkflowSarek.retrieveInput(params, log), checkIfExists: true))
+ch_from_samplesheet = params.build_only_index ? Channel.empty() : Channel.fromSamplesheet("input")
+
+ch_from_samplesheet
+.map{ meta, fastq_1, fastq_2, table, cram, crai, bam, bai, vcf, variantcaller ->
+    [ meta.patient + meta.sample, [meta, fastq_1, fastq_2, table, cram, crai, bam, bai, vcf, variantcaller] ]
+}.tap{ ch_with_patient_sample }
+.groupTuple()
+.map { patient_sample, ch_items ->
+    [ patient_sample, ch_items.size() ]
+}.combine(ch_with_patient_sample, by: 0)
+.map { patient_sample, num_lanes, ch_items ->
+    (meta, fastq_1, fastq_2, table, cram, crai, bam, bai, vcf, variantcaller) = ch_items
+    if (meta.lane && fastq_2) {
+        meta           = meta + [id: "${meta.sample}-${meta.lane}".toString()]
+        def CN         = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+
+        def flowcell   = flowcellLaneFromFastq(fastq_1)
+        // Don't use a random element for ID, it breaks resuming
+        def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+        println "Number of lanes: $num_lanes"
+
+        meta           = meta + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'fastq', size: 1]
+
+        if (params.step == 'mapping') return [ meta - meta.subMap('lane'), [ fastq_1, fastq_2 ] ]
```

Comment:
@FriederikeHanssen don't we need meta.lane at this step still?

Reply:
```suggestion
        if (params.step == 'mapping') return [ meta, [ fastq_1, fastq_2 ] ]
```